library/api/src/main/java/com/epmedu/animeal/api/extensions/Api.kt

```diff
@@ -0,0 +1,81 @@
+package com.epmedu.animeal.api.extensions
+
+import com.amplifyframework.api.ApiException
+import com.amplifyframework.api.aws.GsonVariablesSerializer
+import com.amplifyframework.api.graphql.SimpleGraphQLRequest
+import com.amplifyframework.api.graphql.model.ModelQuery
+import com.amplifyframework.core.Amplify
+import com.amplifyframework.core.model.Model
+import com.amplifyframework.core.model.query.predicate.QueryPredicate
+import com.apollographql.apollo.api.Mutation
+import com.apollographql.apollo.api.Operation.Data
+import com.apollographql.apollo.api.Operation.Variables
+import com.epmedu.animeal.api.wrapper.ApiResult
+import com.epmedu.animeal.api.wrapper.ResponseError
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.channels.awaitClose
+import kotlinx.coroutines.channels.trySendBlocking
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.callbackFlow
+import kotlinx.coroutines.suspendCancellableCoroutine
+import java.util.concurrent.CancellationException
+
+/**
+ * Creates a query for list of models of type [GraphQLModel] with provided [predicate]
+ * and on successful responses returns flow with the list of the models.
+ * On failure flow closes.
+ * @param predicate predicate for filtering. By default `null`
+ */
+internal inline fun <reified GraphQLModel : Model> getModelList(
```

Comment:
I'm not sure if we will have a use case for this, but it will be good to have generic method for queries as well (similar to mutations).

Reply:
Let me do it in the scope of the subscriptions task